### ISSUE TITLE
Why the loss type is margin_loss in untargeted attack, but cross_entropy loss in targeted attack.

### ISSUE THREAD

**Issue by machanic** [max-andr/square-attack#4]
The code of  `args.loss = 'margin_loss' if not args.targeted else 'cross_entropy'` is located in https://github.com/max-andr/square-attack/blob/master/attack.py#L282
Why the loss type is margin_loss in untargeted attack, but cross_entropy loss in targeted attack?

**Comment by fra31** [comment]
Hi!

The margin loss has a straightforward interpretation in terms of the classification (in our case an adversarial example is found when `margin_loss < 0`), then we use it for untargeted attacks.
However, we observed that, in the case of targeted attacks, especially in datasets like ImageNet with many classes, the cross-entropy loss leads to better query efficiency, since it involves the scores of all the classes in its formulation (see Sec. E.1 [here](https://arxiv.org/pdf/1912.00049.pdf) for a discussion of this). Then we use it for the targeted scenario.

Hope this helps!

**Comment by max-andr** [comment]
Closing now. Feel free to reopen in case you want further clarifications!
